### PR TITLE
fixed bugs towards ITF1788 compliance

### PR DESCRIPTION
**Pull request by lucaferranti**
* fixed typo in pow_rev2 (lob -> log
* fixed power_rev for exponent zero integer
* added support for decorated intervals
* added methods with one less parameter with x=R
* fixed cosh_rev
* added mul_rev_to_pair
* added tests from ITF1788 test suite

remaining issues

**ieeep1788_rev.itl**
- 16 tests with power_rev failing. From what I can tell, the returned result is just a few ulps wider than the expected one.
- ~33 tests failing with `mul_rev_IEEE1788`, but haven't had time to look at those yet~

**pow_rev.itl**
- at the moment 233/806 tests failing. Probably the current implementations do not take into account some corner cases? Need to check those

### CODE REVIEW

**Comment by lucaferranti** [comment]
@lbenet most ITF1788 tests were failing for reverse functions, this PR fixes most of them. The remaining one are other tedious (rev_pow, there's a whole phd dissertation on how to compute that) or probably difficult (power_rev in some cases returns slightly wider results). I suggest to merge this now and track the remaining issues in an issue for later.

(As I briefly mentioned yesterday, I think going through rev_pow and implement it could be a suitable project for an undergrad with some interest in interval methods)

**Comment by lbenet** [comment]
@lucaferranti I'll try to find some space this afternoon to review this PR; today I have a very busy morning...

Aside from that, wouldn't this be better in IntervalAithmetic.jl package, precisely for compliance with the standard? This package at the end relies on IA, so there is no new dependency.

**Comment by lucaferranti** [comment]
I don't have an opinion, about whether the reverse functions should be here or in IA, from a standard compliance it may make more sense to have those in IA. Alternatively, one can sell ValidatedNumerics as "the" package.

One thing that could be nice to do in this package is to define a common contractor interface for both IntervalRootFinding.jl and IntervalConstraintProgramming, e.g. currently both packages export `Contractor`, which is a little suboptimal.

**Review by lbenet — COMMENTED** [review]
I've left essentially two comments:
- one related to the generation of the rev functions when the *last interval argument* omitted (is optional), which I think should include *all* the fnctions in the compacted code
- one (actually two) related with the docstrings

Since this PR impacts (I think) https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/505, I suggest to take into account the first comment and merge this PR, and leave the second comment to another PR. Once said this, apart from that comment, LGTM.

File: src/decorated.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
+entiredecorated(T) = DecoratedInterval(entireinterval(T))
+
+for op in (:sqr_rev, :abs_rev, :sin_rev, :cos_rev, :tan_rev, :cosh_rev, :sinh_rev, :tanh_rev)
```
[the remark this answers is not in the document]
Comment (lucaferranti) [reply src/decorated.jl:3]: I'm not sure I get what you mean, the type signature of `power_rev` is different from the one in the methods generated in the loop.

**Comment by lucaferranti** [comment]
@lbenet I've updated some docstrings (the three for powers to disambiguate and a few more), if you think the template is good, I can add to the others too (maybe in a separate PR, if the idea was to merge this before the other PR in IntervalArithmetic.jl).

I didn't quite understand your other comment, the `power_rev` function with one parameter seems to be defined [here](https://github.com/JuliaIntervals/IntervalContractors.jl/pull/49/files#diff-c2a6b6511d82c3a177e374ed5042df1f0539aacf6e6739a9510b64eb272ca4a2R163)

File: src/decorated.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (lbenet) [reply src/decorated.jl:3]: Sorry for the noise. You are right, the signature is different.

**Comment by lbenet** [comment]
In view of https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/271, my only suggestion is to add the reference section of the standard where the rev functions are described; I think this is Sect 10.5.4 and Table 10.1. But this can be included in a separated PR.

**Comment by lucaferranti** [comment]
@lbenet apologies for my delay in coming back to this.

I have now clarified the docstrings for the functions with reference to the standard. Let me know if you want me to do other changes, otherwise once this is merged I can update https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/505

**Review by lbenet — APPROVED** [review]
LGTM! In another PR we may add more descriptive docstrings (with regards to all rev functions and their mention in the standard). Thanks a lot!

**Comment by lbenet** [comment]
Please, go ahead and merge this!
